### ISSUE TITLE
When `Include.NON_DEFAULT` setting is used on POJO, empty values are not included in json if default is `null`

### ISSUE THREAD

**Issue by ragnhov** [FasterXML/jackson-databind#4741]
### Search before asking

- [X] I searched in the [issues](https://github.com/FasterXML/jackson-databind/issues) and found nothing similar.

### Describe the bug

After upgrading to 2.18.0, empty Strings/objects are no longer included in json output if NON_DEFAULT annotation is used on the class and the default value is null. This is possibly related to fixing https://github.com/FasterXML/jackson-databind/issues/4464
As far as I can understand, the documentation states that the test below should pass, as it does in 2.17.1.
![image](https://github.com/user-attachments/assets/47c4fead-c6bb-4c3e-b01f-a857c09fb27c)

### Version Information

2.18.0

### Reproduction

```java
@Test
void testSerialization() throws JsonProcessingException {
    MyString input = new MyString();
    input.setValue("");
    ObjectMapper objectMapper = new ObjectMapper();
    String json = objectMapper.writeValueAsString(input);
    MyString output = objectMapper.readValue(json, MyString.class);
    assertEquals(input.getValue(), output.getValue());
}

@JsonInclude(JsonInclude.Include.NON_DEFAULT)
@Data
public static class MyString {
    private String value = null;
}
```

**Comment by pjfanning** [comment]
That annotation has this in the javadoc:

> When used for a POJO, definition is that only values that differ from the default values of POJO properties are included.

I don't think this annotation value is about nulls - there are other annotation values you can use. If you look at the test cases in jackson-databind, you will see that none of the tests related to `Include.NON_DEFAULT` are about null handling.

Have a look at https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-annotations/latest/index.html

You could try `@JsonInclude(JsonInclude.Include.ALWAYS)`. This seems to be closer to what you want.

**Comment by pjfanning** [comment]
I created #4743 which does demo your issue. I have verified that 2.18 behaves differently.

**Comment by pjfanning** [comment]
Reasonably good chance that this relates to #4464

**Comment by JooHyukKim** [comment]
Tough call...  seems like everyone has their own interpretation of the JavaDoc.
@teodord Could you help us out here? This issue here is sort of contracting to what we agreed to.
Your opinion would matter, be welcome as well

**Comment by ragnhov** [comment]
The documentation states that the behavior depends on context. I believe that the context if different in this case and @teodord 's case. The POJO case should include empty strings, the global case should not.

**Comment by JooHyukKim** [comment]
Okay, re-reading through the documentation I am leaning toward towards keeping both behavior, thus #4744.

I would like to ask @cowtowncoder for direction we should head, while I am going back to re-examine the documentation-behavior sync one more time.

**Comment by cowtowncoder** [comment]
On this particular case: in this example if POJO has `null` as value for `String`, then `null` AND ONLY `null` should be excluded, not empty String. So I think this is valid bug report indeed.

**Comment by cowtowncoder** [comment]
Fixed for 2.18.1
